api/provider/aws/service_create.go

```diff
@@ -76,18 +126,49 @@ func (s *ServiceProvider) Create(req models.CreateServiceRequest) (string, error
 }
 
 func (s *ServiceProvider) createService(
-	cluster string,
-	serviceName string,
-	taskDefinition string,
+	cluster,
+	serviceType,
+	serviceName,
+	taskDefinitionARN,
+	loadBalancerRole,
+	networkMode,
+	fargatePlatformVersion string,
 	desiredCount int,
-	loadBalancerRole string,
+	subnets []string,
+	securityGroupIDs []*string,
 	loadBalancer *ecs.LoadBalancer,
 ) error {
 	input := &ecs.CreateServiceInput{}
 	input.SetCluster(cluster)
 	input.SetDesiredCount(int64(desiredCount))
 	input.SetServiceName(serviceName)
-	input.SetTaskDefinition(taskDefinition)
+	input.SetTaskDefinition(taskDefinitionARN)
+
+	launchType := ecs.LaunchTypeEc2
+	if serviceType == models.DeployCompatibilityStateless {
+		launchType = ecs.LaunchTypeFargate
+		input.SetPlatformVersion(fargatePlatformVersion)
+	}
+
+	input.SetLaunchType(launchType)
+
+	if networkMode == "awsvpc" {
```

Comment:
is there a `const` in `aws-sdk-go` we can user here?

Reply:
Hot dang, there is! Great call. ✅